crates/server/src/store/file.rs

```diff
@@ -93,6 +115,29 @@ impl FileAuthority {
         root_dir: Option<&Path>,
         config: &FileConfig,
         #[cfg(feature = "__dnssec")] nx_proof_kind: Option<NxProofKind>,
+    ) -> Result<Self, String> {
+        Self::try_from_config_internal(
+            origin,
+            zone_type,
+            allow_axfr,
+            root_dir,
+            config,
+            #[cfg(feature = "__dnssec")]
+            nx_proof_kind,
+            #[cfg(feature = "metrics")]
+            false,
+        )
+    }
+
+    // internal load for e.g. sqlite db creation
+    pub(crate) fn try_from_config_internal(
```

Comment:
It seems pretty ugly to add both `inner_new()` and `try_from_config_internal()`. Can we simplify?

Reply:
Re-factored, removed the `inner_new()`.